docs/standard/serialization/system-text-json-converters-how-to.md

```diff
@@ -0,0 +1,755 @@
+* Override the `Write` method to serialize the incoming object of type `T`. Use the <xref:System.Text.Json.Utf8JsonWriter> that is passed to the method to write the JSON.
+* Override the `CanConvert` method only if necessary. The default implementation returns `true` when the type to convert is type `T`. Therefore, converters that support only type `T` don't need to override this method. For an example of a converter that does need to override this method, see the [polymorphic deserialization](#support-polymorphic-deserialization) section later in this article.
+
+You can refer to the [built-in converters source code](https://github.com/dotnet/corefx/tree/master/src/System.Text.Json/src/System/Text/Json/Serialization/Converters/) as reference implementations for writing custom converters.
+
+### The factory pattern
+
+The following steps explain how to create a converter by following the factory pattern:
+
+* Create a class that derives from <xref:System.Text.Json.Serialization.JsonConverterFactory>.
+* Override the `CanConvert` method to return true when the type to convert is one that the converter can handle. For example, if the converter is for `List<T>` it might only handle `List<int>`, `List<string>`, and `List<DateTime>`. 
+* Override the `CreateConverter` method to return an instance of a converter class that will handle the type-to-convert that is provided at runtime.
+* Create the converter class that the `CreateConverter` method instantiates. 
+
+The factory pattern is required for open generics because the code to convert an object to and from a string isn't the same for all types. A converter for an open generic type (`List<T>`, for example) has to create a converter for a closed generic type (`List<DateTime>`, for example) behind the scenes. Code must be written to handle each closed-generic type that the converter can handle.
+
+The `Enum` type is similar to an open generic type: a converter for `Enum` has to create a converter for a specific `Enum` (`WeekdaysEnum`, for example) behind the scenes. 
+
+## Error handling
+
+If you need to throw an exception in error-handling code, consider throwing a <xref:System.Text.Json.JsonException> without a message. This exception type automatically creates a message that includes the path to the part of the JSON that caused the error. For example, the statement `throw new JsonException();` produces an error message like the following example:
+
+```text
+Unhandled exception. System.Text.Json.JsonException: 
+The JSON value could not be converted to System.Object. 
+Path: $.Date | LineNumber: 1 | BytePositionInLine: 37.
+```
+
+If you do provide a message (for example, `throw new JsonException("Error occurred)`, the exception still provides the path in the <xref:System.Text.Json.JsonException.Path> property.
+
+### Sample basic converter
```

Comment:
It seems we should lead with this and then explain the factory pattern. It feels strange to "force" the customer to read a lot of conceptual text before they get to some sample code.